compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -437,15 +548,48 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       if (shouldCheckLiterals)
         checkJSNameLiteral(sym)
 
+      if (enclosingOwner is OwnerKind.JSClass) {
+        // Private methods cannot be overloaded
+        if (sym.isMethod && isPrivateMaybeWithin(sym)) {
+          val alts = sym.owner.info.member(sym.name).filter(_.isMethod)
+          if (alts.isOverloaded) {
+            reporter.error(tree.pos,
+                "Private methods in Scala.js-defined JS classes cannot be " +
+                "overloaded. Use different names instead.")
+          }
+        }
+
+        // private[Scope] methods must be final
+        if (sym.isMethod && (sym.hasAccessBoundary && !sym.isProtected) &&
+            !sym.isFinal) {
+          reporter.error(tree.pos,
+              s"The qualified private method ${sym.nameString} in a " +
+              "Scala.js-defined JS class must be final")
```

Comment:
That is a strange message. To me it means, that only the method with this name must be final... Others are OK. IMO there is no need to put the name in the method... The error will indicate the right location.